libnavigation-core/src/main/java/com/mapbox/navigation/core/history/HistoryFiles.kt

```diff
@@ -0,0 +1,48 @@
+package com.mapbox.navigation.core.history
+
+import android.content.Context
+import com.mapbox.base.common.logger.Logger
+import com.mapbox.base.common.logger.model.Message
+import com.mapbox.base.common.logger.model.Tag
+import com.mapbox.navigation.base.options.HistoryRecorderOptions
+import java.io.File
+
+/**
+ * Create a default directory for the history files to be saved.
+ * When a directory has been provided through options, ensure it exists.
+ */
+internal class HistoryFiles(
+    val applicationContext: Context,
+    val logger: Logger
+) {
+
+    fun absolutePath(options: HistoryRecorderOptions): String? {
+        val fileDirectory = options.fileDirectory ?: defaultFilePath()
+        val historyFile = File(fileDirectory)
+        if (!historyFile.exists()) {
+            historyFile.mkdirs()
+        }
+        return if (historyFile.exists()) {
+            logger.i(loggerTag, Message("Initial size is ${historyFile.length()} bytes"))
+            historyFile.absolutePath
+        } else {
+            logger.e(
+                loggerTag,
+                Message(
+                    "Unable to create a file, check the HistoryRecorderOptions " +
+                        historyFile.absolutePath
+                )
+            )
+            null
+        }
+    }
+
+    private fun defaultFilePath(): String {
+        return File(applicationContext.filesDir, TILES_PATH_SUB_DIR).absolutePath
+    }
+
+    companion object {
```

Comment:
```suggestion
    internal companion object {
```

Reply:
class is internal so it isn't needed

Reply:
In any case, it seems it can be `private` as it's only used within the class so maybe we can implement

```kotlin
private companion object {
        private val loggerTag = Tag("MbxHistoryRecorderOptions")
        private const val TILES_PATH_SUB_DIR = "mbx_nav/history"
}
```

Reply:
This https://github.com/mapbox/mapbox-navigation-android/pull/4488/files#r650094270 still applies.